### PR TITLE
perf(ext/websocket): optimize socket.send

### PR DESCRIPTION
**Pull request by littledivy**
Towards #16315

### CODE REVIEW

File: ext/websocket/lib.rs [hunk 1]
```diff
@@ -426,6 +471,54 @@ pub async fn op_ws_send(
...
+    .get::<WsStreamResource>(rid)?;
+  resource.send(Message::Binary(data.to_vec())).await?;
+  Ok(())
+}
+
+#[op]
+pub fn op_ws_try_send_string(
+  state: &mut OpState,
+  rid: ResourceId,
+  text: String,
```
Comment (aapoalas) [inline ext/websocket/lib.rs:506]: Would it make sense to make this op an `#[op(v8)]`, pass text in as a `serde_v8::Value` and then use the `v8::String::write_utf8` method to write directly into the sink? (I'm not sure if that's actually possible either.)

Comment (aapoalas) [reply ext/websocket/lib.rs:506]: Nevermind, just noticed that the `Message` type actually requires a String so this is as optimal as one can get I guess.

File: ext/websocket/lib.rs [hunk 2]
```diff
@@ -426,6 +471,54 @@ pub async fn op_ws_send(
...
+}
+
+#[op(fast)]
+pub fn op_ws_try_send_binary(
+  state: &mut OpState,
+  rid: u32,
+  value: &[u8],
+) -> Result<bool, AnyError> {
+  let resource = state.resource_table.get::<WsStreamResource>(rid)?;
+  resource.try_send(Message::Binary(value.to_vec()))
```
Comment (aapoalas) [inline ext/websocket/lib.rs:519]: Sad that it's not possible to just use the slice for this synchronous call :/

**Nit:** There's a `From<&'b [u8]>` implementation for `Message`.

Comment (aapoalas) [reply ext/websocket/lib.rs:519]: Also a `From<String>` apparently.

Comment (littledivy) [reply ext/websocket/lib.rs:519]: oh its indeed possible to assume `'static` lifetime for that slice since this is a sync call.

Comment (aapoalas) [reply ext/websocket/lib.rs:519]: Oh yeah. I couldn't find a method for it directly but I guess it should be possible to just do:
```rs
let vec = Vec::from_raw_parts::<u8>(value.as_ptr(), value.len(), value.len());
```
and avoid the copy.

Comment (littledivy) [reply ext/websocket/lib.rs:519]: umm we cant do that. tungstenite will free the vec owned by v8

blocked on https://github.com/snapview/tungstenite-rs/pull/175

Comment (aapoalas) [reply ext/websocket/lib.rs:519]: Oh very true. Note to self: Don't lie to Rust about ownership of structs.

**Review by crowlKats — APPROVED** [review]

**Pull request merged**